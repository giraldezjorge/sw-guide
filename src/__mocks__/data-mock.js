
export const mockCharacter = {
  name: 'Luke Skywalker',
  height: '172',
  mass: '77',
  hair_color: 'blond',
  skin_color: 'fair',
  eye_color: 'blue',
  birth_year: '19BBY',
  gender: 'male',
  url: 'test/people/1/'
}

export const mockCharacters = [
  mockCharacter,
  {
    name: 'C-3PO',
    height: '167',
    mass: '75',
    hair_color: 'n/a',
    skin_color: 'gold',
    eye_color: 'yellow',
    birth_year: '112BBY',
    gender: 'n/a',
    url: 'test/people/2/'
  },
  {
    name: 'R2-D2',
    height: '96',
    mass: '32',
    hair_color: 'n/a',
    skin_color: 'white, blue',
    eye_color: 'red',
    birth_year: '33BBY',
    gender: 'n/a',
    url: 'test/people/3/'
  }
]

export const mockFilm = {
  title: 'A New Hope',
  episode_id: 4,
  opening_crawl: 'It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire\'s\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire\'s\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy....',
  director: 'George Lucas',
  producer: 'Gary Kurtz, Rick McCallum',
  release_date: '1977-05-25',
  url: 'test/films/1/',
  characters: [
    'test/people/1/'
  ]
}

export const mockFilms = [
  mockFilm,
  {
    title: 'The Empire Strikes Back',
    episode_id: 5,
    opening_crawl: 'It is a dark time for the\r\nRebellion. Although the Death\r\nStar has been destroyed,\r\nImperial troops have driven the\r\nRebel forces from their hidden\r\nbase and pursued them across\r\nthe galaxy.\r\n\r\nEvading the dreaded Imperial\r\nStarfleet, a group of freedom\r\nfighters led by Luke Skywalker\r\nhas established a new secret\r\nbase on the remote ice world\r\nof Hoth.\r\n\r\nThe evil lord Darth Vader,\r\nobsessed with finding young\r\nSkywalker, has dispatched\r\nthousands of remote probes into\r\nthe far reaches of space....',
    director: 'Irvin Kershner',
    producer: 'Gary Kurtz, Rick McCallum',
    release_date: '1980-05-17',
    url: 'test/films/2/',
    characters: [
      'test/people/1/'
    ]
  }
]
