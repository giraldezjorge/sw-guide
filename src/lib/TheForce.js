import axios from 'axios'
import { API_URL } from '../constants/api'

const TheForce = function () {
  if (!(this instanceof TheForce)) return new TheForce()
}

TheForce.prototype.getFilms = function () {
  return axios.get(API_URL + 'films')
    .then((films) => films.data.results)
    .then((films) => films.sort((a, b) => {return new Date(a.release_date) - new Date(b.release_date)}))
    .catch((error) => console.log(error))
}

TheForce.prototype.getFilm = function (id) {
  return axios.get(API_URL + 'films/' + id)
    .then((film) => film.data)
    .catch((error) => console.log(error))
}

TheForce.prototype.getCharacter = function (id) {
  return axios.get(API_URL + 'people/' + id)
    .then((character) => character.data)
    .catch((error) => console.log(error))
}

export default TheForce
