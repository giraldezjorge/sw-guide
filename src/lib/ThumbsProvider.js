import sw_1 from '../assets/sw_1.jpg'
import sw_2 from '../assets/sw_2.jpg'
import sw_3 from '../assets/sw_3.jpg'
import sw_4 from '../assets/sw_4.jpg'
import sw_5 from '../assets/sw_5.jpg'
import sw_6 from '../assets/sw_6.jpg'
import sw_7 from '../assets/sw_7.jpg'
import sw_8 from '../assets/sw_8.jpg'
import sw_9 from '../assets/sw_9.jpg'

const ThumbsProvider = function () {
  if (!(this instanceof ThumbsProvider)) return new ThumbsProvider()
}

ThumbsProvider.prototype.getThumb = function (ep_id) {
  const thumbs = [sw_1, sw_2, sw_3, sw_4, sw_5, sw_6, sw_7, sw_8, sw_9]
  return thumbs[ep_id] || thumbs[thumbs.length - 1]
}

export default ThumbsProvider
