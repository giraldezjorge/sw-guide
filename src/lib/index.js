
exports.theForce = require('./TheForce').default()
exports.thumbsProvider = require('./ThumbsProvider').default()
