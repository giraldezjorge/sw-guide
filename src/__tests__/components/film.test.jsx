import React from 'react'
import mockAxios from 'axios'
import { mockFilm, mockCharacter } from '../../__mocks__/data-mock'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { mount } from 'enzyme'
import Film from '../../components/film'
import { theForce } from '../../lib'

const mockStore = configureStore([])

describe('The film component', () => {
  const film = mockFilm

  const store = mockStore({ favorites: [] })

  mockAxios.get.mockResolvedValue({
    data: film
  })

  afterEach(() => {
    mockAxios.get.mockClear()
  })

  it('should fetch film data', () => {
    const getSpy = jest.spyOn(theForce, 'getFilm')
    const filmInstance = mount(
      <MemoryRouter>
        <Provider store={store}>
          <Film film={film} params={{ id: 1 }} />
        </Provider>
      </MemoryRouter>
    )

    expect(getSpy).toBeCalled()
  }),

  it('shows the film info', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Film film={film} params={{ id: 1 }} />
        </Provider>
      </MemoryRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
