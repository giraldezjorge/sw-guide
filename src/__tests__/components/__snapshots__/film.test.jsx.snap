// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The film component shows the film info 1`] = `null`;
